app/models/user.rb

```diff
@@ -125,6 +126,19 @@ def set_avatar_from_twitter
   rescue
   end
 
+  # Extract the information that is available after OmniAuth authentication.
+  def self.from_omniauth(auth, params)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.provider = auth.provider
```

Comment:
@stedmanblake made some updates / changes on a separate branch if you want to check it out: PR #29